### DISCUSSION TITLE
Does the .env support basically replace the functionality in next.config.js?

### DISCUSSION THREAD

**Discussion by pkellner** [vercel/next.js#20857] [Help] [answered]
For just setting environmental variables where no JavaScript processing code is needed, is it better to just use the new Environment Variable features and get rid of all of that in next.config.js?  Seems like there is a bunch of overlap and the guidance is not clear to me.

https://nextjs.org/docs/basic-features/environment-variables

**Comment by jamesmosier** [comment, marked as answer]
Yes, the recommendation is to use `.env` files over the `env` object in `next.config.js`. On the old [next.config.js env page](https://nextjs.org/docs/api-reference/next.config.js/environment-variables) it is mentioned that the new `.env` file method is "more intuitive and ergonomic" which leads me to believe that it is preferred.
